bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Font.java

```diff
@@ -195,7 +213,18 @@ public FontData[] getFontData() {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
 	LOGFONT logFont = new LOGFONT ();
 	OS.GetObject(handle, LOGFONT.sizeof, logFont);
-	return new FontData[] {FontData.win32_new(logFont, device.computePoints(logFont, handle))};
+	float heightInPoints = device.computePoints(logFont, handle);
+	int primaryZoom = DPIUtil.mapDPIToZoom (device._getDPIx());
+	float adjustedZoomFactor;
+	if (zoomFactor != primaryZoom) {
+		// as Device::computePoints will always return point on the basis of the
+		// primary monitor zoom, a custom zoomFactor must be calculated if the font
+		// is used for a different zoom level
+		adjustedZoomFactor = 1.0f * primaryZoom / zoomFactor;
+	} else {
+		adjustedZoomFactor = 1.0f;
+	}
```

Comment:
This seems to be a premature optimization (assuming `zoomFactor != 0`) , or do I miss something?
```suggestion
	float adjustedZoomFactor = 1.0f * primaryZoom / zoomFactor;
```